cpp/include/cugraph/algorithms.hpp

```diff
@@ -1677,6 +1679,99 @@ void triangle_count(raft::handle_t const& handle,
                     raft::device_span<edge_t> counts,
                     bool do_expensive_check = false);
 
+/**
+ * @brief     Compute Jaccard similarity coefficient
+ *
+ * Similarity is computed based for every pair of vertices specified, or (if no vertices
+ * are specified then all vertices that are 2 hops apart are computed).  Note that
+ * similarity algorithms expect a symmetric graph.
+ *
+ * @throws                 cugraph::logic_error when an error occurs.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @tparam edge_t Type of edge identifiers. Needs to be an integral type.
+ * @tparam weight_t Type of edge weights. Needs to be a floating point type.
+ * @tparam multi_gpu Flag indicating whether template instantiation should target single-GPU (false)
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Graph view object.
+ * @param first optional device span defining the first vertex in the desired vertex pairs
+ * @param second optional device span defining the second vertex in the desired vertex pairs
```

Comment:
Any requirements about pair shuffling in MG?

I guess we need a bit more descriptive variable name than `first` and `second`... Maybe something like `vertex_pair_firsts` and `vertex_pair_seconds`?

Reply:
I now feel like `std::tuple<raft::device_span<vertex_t const>, raft::device_span<vertex_t const>> vertex_pairs` sounds like a better name.